dao/service.go

```diff
@@ -0,0 +1,20 @@
+package dao
+
+import "github.com/companieshouse/emergency-auth-code-api/config"
+
+// Service interface declares how to interact with the persistence layer regardless of underlying technology
+type Service interface {
```

Comment:
I would rename this to something more explicit like `UserAuthDatabase` rather than `Service` as we're gonna have another database we're connecting to with Oracle. That way we can have two interfaces in this package, one for connecting to the Auth database (with a mongo implementation that's done here) and one for connecting to the directors database (with an oracle implementation) within this same file.